docs/csharp/whats-new/csharp-9.md

```diff
@@ -0,0 +1,233 @@
+    {
+        static void Main(string[] args)
+        {
+            Console.WriteLine("Hello World!");
+        }
+    }
+}
+```
+
+There’s only one line of code that does anything. With top-level statements, you can replace all that boilerplate with the `using` statement and the single line that does the work:
+
+:::code language="csharp" source="snippets/whats-new-csharp9/Program.cs" ID="TopLevelStatements":::
+
+Only one file in your application may use top-level statements. If the compiler finds top-level statements in multiple source files, it’s an error. It’s also an error if you combine top-level statements with a declared program entry point method, typically a `Main` method. In a sense, you can think that one file contains the statements that would normally be in the `Main` method of a `Program` class.  
+
+One of the most common uses for this feature is creating teaching materials. Beginner C# developers can write the canonical “Hello World!” in two lines of code. None of the extra ceremony is needed. However, seasoned developers will find many uses for this feature as well. Top-level statements enable a script-like experience for experimentation similar to what Jupyter notebooks provide. Top-level statements are great for small console programs and utilities. In addition, Azure functions are an ideal use case for top-level statements.
+
+Most importantly, top-level statements do not limit your application’s scope or complexity. Those statements can access or use any .NET class. They also don’t limit your use of command-line arguments or return values. Top-level statements an access an array or strings named args. If the top-level statements return an integer value, that value becomes the integer return code from a synthesized `Main` method. The top-level statements may contain async expressions. In that case, the synthesized entry point returns a `Task`, or `Task<int>`.
+
+## Pattern matching enhancements
+
+C# 9 includes new pattern matching improvements:
+
+- ***Type patterns*** match a variable is a type
+- ***Parenthesized patterns*** enforce or emphasize the precedence of pattern combinations
+- ***Conjunctive `and` patterns*** require both patterns to match
+- ***Disjunctive `or` patterns*** require either pattern to match
+- ***Negated `not` patterns*** require that a pattern doesn’t match
+- ***Relational patterns*** require the input be less than, greater than, less than or equal, or greater than or equal to a given constant.
+
+These patterns enrich the syntax for patterns. Consider these examples:
+
+:::code language="csharp" source="snippets/whats-new-csharp9/PatternUtilities.cs" ID="IsLetterPattern":::
+
+Alternatively, with optional parentheses to make it clear that and has higher precedence than or:
+
+:::code language="csharp" source="snippets/whats-new-csharp9/PatternUtilities.cs" ID="IsLetterOrSeparatorPattern":::
+
+One of the most common uses is a new syntax for a null check:
+
+```csharp
+if (e is not null)
```

Comment:
Should you include previous ways that this was checked?

```csharp
if (!(e is null))
```

or 

```csharp
if (e != null)
```

Reply:
I thought about both of those.  There were so many ways to do null checks, I didn't want to list them. Consider these alternatives:
```csharp
// 1. 
_ = e ?? throw null;
// 2. 
if (e is object)
```

As for the operator overloading, I'll add that to my list for updating `null` check guidance for C# 9. (I'll be creating issues later today.)